operating_versioned_docs/version-v0.5/manuals/poddecoration.md

```diff
@@ -0,0 +1,352 @@
+  - affectedReplicas: 3
+    collaSet: foo
+    pods:
+    - name: foo-2gnnl
+      revision: sample-pd-8697d4bf8c
+    - name: foo-lftw8
+      revision: sample-pd-9465f4c84
+    - name: foo-n57rr
+      revision: sample-pd-9465f4c84
+  matchedPods: 3
+  injectedPods: 3
+  updatedPods: 1
+  updatedReadyPods: 1
+  updatedAvailablePods: 1
+  isEffective: true
+  currentRevision: sample-pd-9465f4c84
+  updatedRevision: sample-pd-8697d4bf8c
+```
+
+
+# Features
+
+## Injection
+
+### Metadata
+```yaml
+apiVersion: apps.kusionstack.io/v1alpha1
+kind: PodDecoration
+spec:
+  template:
+    metadata:
+    - patchPolicy: MergePatchJson
+      annotations:
+        cafe.sofastack.io/decoration-version: '[{"name":"sample-pd","version":"v2"}]'
+    - patchPolicy: Overwrite
+      labels:
+        custom.io/sidecar-version: "v2"
+      annotations:
+        cafe.sofastack.io/decoration-name: sample-pd
+```
+`patchPolicy` is the injected policy, as follows:
+- `Retain`: The original value of annotations and labels will be retained.
+- `Overwrite`: The value of annotations and labels corresponding to the existing key will be overwritten.
+- `MergePatchJson`: It only takes effect for annotation. If the key does not exist, the value will be written directly. Otherwise, the json value will be merged.
+
+For example：
+```yaml
+# Old pod metadata
+metadata:
+  labels:
+    custom.io/sidecar-version: "v1"
+  annotations:
+    cafe.sofastack.io/decoration-version: '[{"name":"old-pd","version":"v1"}]'
+
+# After metadata injected
+metadata:
+  labels:
+    custom.io/sidecar-version: "v2"
+  annotations:
+    cafe.sofastack.io/decoration-type: sample-pd
+    cafe.sofastack.io/decoration-version: '[{"name":"old-pd","version":"v1"}, {"name":"sample-pd","version":"v2"}]'
+```
+### Primary Container
+
+```yaml
+apiVersion: apps.kusionstack.io/v1alpha1
+kind: PodDecoration
+spec:
+  template:
+    primaryContainers:
+    - targetPolicy: ByName
+      name: foo
+      image: foo:v2
+      env: 
+      - name: APP_NAME
+        value: foo
+      volumeMounts:
+      - name: sample-volume
+        mountPath: /vol/sample
+    volumes:
+    - name: sample-volume
+      emptyDir: {}
+```
+Injection into the primary containers only supports limited fields: `image`, `env` and `volumeMounts`.
+
+`targetPolicy` indicates which existed container these configuration should inject into, as follows:
+- `ByName`: Only inject containers matching `name`.
+- `All`: Inject all primary containers.
+- `First`: Inject into first primary container.
+- `Last`: Inject into last primary container.
+
+### Sidecar Container
+
+```yaml
+spec:
+  template:
+    containers:
+    - injectPolicy: AfterPrimaryContainer  # Container injected policy, AfterPrimaryContainer or BeforePrimaryContainer
+      name: sidecar-a
+      image: ubuntu:22.04
+      ...
+```
+Inject a new sidecar container. Optional, it can be placed in front or behind the primary container.
+### InitContainer
+
+```yaml
+spec:
```

Comment:
some pd demo has apiversion/name and some not, maybe keep them same?

Reply:
this will be fixed